.github/workflows/main.yaml

```diff
@@ -16,24 +16,21 @@ jobs:
         with:
           fetch-depth: 1
       - name: Setup Ruby
-        uses: ruby/setup-ruby@55283cc23133118229fd3f97f9336ee23a179fcf # v1.146.0
+        uses: ruby/setup-ruby@97e35c5302afcf3f5ac1df3fca9343d32536b286 # v1.184.0
         with:
-          ruby-version: '3.1' # Not needed with a .ruby-version file
+          ruby-version: '.ruby-version' # Not needed with a .ruby-version file
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
           cache-version: 0 # Increment this number if you need to re-download cached gems
       - name: Make sure node is installed.
         run: sudo apt-get update && sudo apt-get -y install nodejs
       - name: Install md-to-pdf (NodeJS)
-        run: npm i -g md-to-pdf
+        run: corepack npm i -g md-to-pdf
```

Comment:
Corepack is Node's package manager manager and may be more up-to-date and have more consistent behaviour than random offered here.